dspace-api/src/main/java/org/dspace/ctask/general/RegisterDOI.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.ctask.general;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.curate.AbstractCurationTask;
+import org.dspace.curate.Curator;
+import org.dspace.identifier.DOIIdentifierProvider;
+import org.dspace.identifier.IdentifierException;
+import org.dspace.identifier.doi.DOIIdentifierNotApplicableException;
+import org.dspace.utils.DSpace;
+
+/**
+ * This curation task will register a DOI for an item, optionally ignoring any logical filtering applied
+ * to normal identifier registration and DOI service operation.
+ *
+ * @author Kim Shepherd
+ */
+public class RegisterDOI extends AbstractCurationTask {
+    // Curation task status
+    private int status = Curator.CURATE_SUCCESS;
+    // The skipFilter boolean has a default value of 'true', as per intended operation
+    private boolean skipFilter = true;
+    // The distributed boolean has a default value of 'false' for safest operation
+    private boolean distributed = false;
+    // Prefix for configuration module
+    private static final String PLUGIN_PREFIX = "doi-curation";
+    // Logger
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(RegisterDOI.class);
+    // DOI provider
+    private DOIIdentifierProvider provider;
+
+    /**
+     * Initialise the curation task and read configuration, instantiate the DOI provider
+     */
+    @Override
+    public void init(Curator curator, String taskId) throws IOException {
+        super.init(curator, taskId);
+        // Get 'skip filter' behaviour from configuration, with a default value of 'true'
+        skipFilter = configurationService.getBooleanProperty(PLUGIN_PREFIX + ".skip-filter", true);
```

Comment:
I would prefer that to default to false

Reply:
I haven't looked deep into this and might be wrong. But I think "true" is the conservative way here. DSpace is not able to filter DOIs so far. The filter is new. Skipping the filter is the old behavior. Do I miss anything?

Reply:
yes @abollini has left a preference here and in the configuration of a default to false.
i do think 'true' is the conservative approach and matches existing DSpace operation.
another way to look at it is "skipFilter = true" is the same as "alwaysApplyFilter = false", if that makes sense?
this option is only used in the CLI, curation task, special minting buttons in the UI which allows administrators to force through DOIs that would otherwise be filtered out in normal operation